Sources/TensorFlow/Core/Tensor.swift

```diff
@@ -559,3 +559,9 @@ extension Tensor: Differentiable where Scalar: TensorFlowFloatingPoint {
     public typealias TangentVector = Tensor
     public typealias AllDifferentiableVariables = Tensor
 }
+
+extension Tensor {
```

Comment:
`withoutDerivative` is available in an extension to `Differentiable`. I'm not sure why this is necessary.

Reply:
If you meant the this is for when `Scalar` is not differentiable, I I think there are other solutions to this. I made a suggestion in a different comment.

Reply:
It's required because if you have a `Tensor<Bool>`, for example (e.g., [here](https://github.com/tensorflow/swift-apis/blob/9e75132b54dcc93dca6dee49466e92b184fc83b2/Sources/TensorFlow/Operators/Math.swift#L434)), then you'd need to wrap it into a thunk to allow the function itself to be differentiable. Is there a nicer way to do this?

Reply:
Nvm, I saw your other comment. :)